src/test/java/com/algorand/algosdk/integration/Stepdefs.java

```diff
@@ -715,23 +820,32 @@ public void sendMsigTxn() throws JsonProcessingException, ApiException{
     }
 
     @Then("the transaction should go through")
-    public void checkTxn() throws ApiException, InterruptedException{
+    public void checkTxn() throws Exception {
         String ans = acl.pendingTransactionInformation(txid).getFrom();
         assertThat(this.txn.sender.toString()).isEqualTo(ans);
-        acl.waitForBlock(lastRound.add(BigInteger.valueOf(2)));
+        waitForAlgodTransactionProcessingToComplete();
         String senderFromResponse = acl.transactionInformation(txn.sender.toString(), txid).getFrom();
         assertThat(senderFromResponse).isEqualTo(txn.sender.toString());
         assertThat(acl.transaction(txid).getFrom()).isEqualTo(senderFromResponse);
     }
 
+    /**
+     * waitForAlgodTransactionProcessingToComplete is a Dev mode helper method that's a rough analog to `acl.waitForBlock(lastRound.add(BigInteger.valueOf(2)));`.
+     * <p>
+     * Since Dev mode produces blocks on a per transaction basis, it's possible algod generates a block _before_ the corresponding SDK call to wait for a block.  Without _any_ wait, it's possible the SDK looks for the transaction before algod completes processing.  So, the method performs a local sleep to simulate waiting for a block.
+     */
+    private static void waitForAlgodTransactionProcessingToComplete() throws Exception {
```

Comment:
Did you try without this function? Seems like these should just work.

Reply:
@winder Resolving with the accompanying explanation - let me know if you'd like to discuss further.

I did try without `waitForAlgodTransactionProcessingToComplete` and the tests consistently fail.  I think the failure stems from the asynchronous timing concerns described in the method's Javadoc.